### DISCUSSION TITLE
How can a file be used instead of a url in the header image?

### DISCUSSION THREAD

**Discussion by gbethardy-h2oai** [h2oai/wave#2190] [Q&A] [answered]
How can a file be used instead of a url in the header image?

**Comment by mturoci** [comment, marked as answer]
Hi @gbethardy-h2oai.

You can also manually upload the image to the server and use that:

```py
from h2o_wave import Q, ui, main, app


@app('/')
async def serve(q: Q):
    q.app.img, = await q.site.upload(['img.png'])
    q.page['header'] = ui.header_card(
        box='1 1 -1 1',
        title='My app',
        subtitle='Let\'s get started!',
        image=q.app.img
    )
    await q.page.save()
```

See [docs](https://wave.h2o.ai/docs/files/#serving-images) for more info.

However, a better option is to usually go for public dir option and use its URL then. See

https://wave.h2o.ai/docs/configuration/#configuring-the-server
https://wave.h2o.ai/docs/configuration/#publicprivate-dirs
https://wave.h2o.ai/docs/files/#serving-files-directly-from-the-wave-server
